test/byContract/IexecPoco/04_finalize.test.ts

```diff
@@ -356,6 +356,150 @@ describe('IexecPoco2#finalize', async () => {
         expect(task.resultsCallback).to.equal('0x'); // deal without callback
     });
 
+    describe('IexecPoco2#finalize-with-scheduler-kitty-part-reward', async () => {
+        [
+            {
+                testInfo:
+                    'scheduler gets 10% of the kitty when these 10% are greater than MIN_KITTY', // (MIN_KITTY=1RLC)
+                workerpoolPriceToFillKitty: 33_333_333_367,
+                expectedKittyBeforeFinalize: 10_000_000_010, // ~10 RLC
+                expectedSchedulerKittyPartReward: 1_000_000_001, // ~1.01 RLC (expectedKittyBeforeFinalize x 10%)
+            },
+            {
+                testInfo: 'scheduler gets at least MIN_KITTY if available',
+                workerpoolPriceToFillKitty: 3_333_333_334,
+                expectedKittyBeforeFinalize: 1_000_000_000, // 1 RLC
+                expectedSchedulerKittyPartReward: 1_000_000_000,
+            },
+            {
+                testInfo: 'scheduler gets all kitty when the kitty is lower than MIN_KITTY',
+                workerpoolPriceToFillKitty: 3_333_333_330,
+                expectedKittyBeforeFinalize: 999_999_999, // ~0.99 RLC
+                expectedSchedulerKittyPartReward: 999_999_999,
+            },
+        ].forEach((testArgs) => {
+            const {
+                testInfo,
+                workerpoolPriceToFillKitty,
+                expectedKittyBeforeFinalize,
+                expectedSchedulerKittyPartReward,
+            } = testArgs;
+            it(`Should finalize task with scheduler kitty part reward where ${testInfo}`, async () => {
+                // Fill kitty
+                const kittyAddress = await iexecPoco.kitty_address();
+                const kittyFillingDealVolume = 1;
+                const kittyFillingSchedulerDealStake = await iexecWrapper.computeSchedulerDealStake(
+                    workerpoolPriceToFillKitty,
+                    kittyFillingDealVolume,
+                );
+                const kittyFillingSchedulerTaskStake =
```

Comment:
we should math.floor this operation then ? since here it's volume = 1

Reply:
In this cas dealStake = taskStake so flooring not needed, do you want me to add it?

Reply:
no : * volume / volume